### PR TITLE
objclass: limit range of no c++ var name mangling

### PR DESCRIPTION
**Pull request by changchengx**
Signed-off-by: Changcheng Liu <changcheng.liu@intel.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by changchengx** [comment]
solve below problem:
```
WARN  2020-05-19 14:44:43,097 [shard 0] osd - _load_class could not open class /home/jenkins-build/build/workspace/ceph-perf-crimson/ceph-master/build/lib/libcls_journal.so (dlopen failed): /home/jenkins-build/build/workspace/ceph-perf-crimson/ceph-master/build/lib/libcls_journal.so: undefined symbol: _Z25cls_get_pool_stripe_widthPv
```

File: src/objclass/objclass.h [hunk 1]
```diff
@@ -4,8 +4,6 @@
 #ifndef CEPH_OBJCLASS_H
 #define CEPH_OBJCLASS_H
 
-#ifdef __cplusplus
```
Comment (tchaikov) [inline src/objclass/objclass.h:7]: why? assume we need to include this header file in a compiling unit programmed in C language, we should exclude the C++ headers from it, right? presumably that's why we add this guard.

File: src/objclass/objclass.h [hunk 2]
```diff
@@ -75,6 +74,7 @@ extern void class_fini(void);
 
 #ifdef __cplusplus
 }
+#endif
```
Comment (tchaikov) [inline src/objclass/objclass.h:77]: ack.

File: src/objclass/objclass.h [hunk 3]
```diff
@@ -162,6 +162,4 @@ bool cls_has_chunk(cls_method_context_t hctx, std::string fp_oid);
 extern uint64_t cls_get_osd_min_alloc_size(cls_method_context_t hctx);
 extern uint64_t cls_get_pool_stripe_width(cls_method_context_t hctx);
 
-#endif
-
-#endif
+#endif // CEPH_OBJCLASS_H
```
Comment (tchaikov) [inline src/objclass/objclass.h:165]: but shall we guard the methods whose names do not start with "cls_cxx", as i think this is convention we need to folow to differentiate the C++ methods and C functions? where the latter can be consumed by rados obj classes written in C language.

Comment (changchengx) [reply src/objclass/objclass.h:165]: I'll follow the convention.

File: src/objclass/objclass.h (hunk 1 above)
Comment (changchengx) [reply src/objclass/objclass.h:7]: You're right. We shouldn't change the scope guard here.

**Comment by changchengx** [comment]
@tchaikov 
We don't need this PR. Below WARN is the "normal" case.
For the other problems exposed in this PR, It depends on the way that how developers use them.
```
WARN  2020-05-19 14:44:43,097 [shard 0] osd - _load_class could not open class /home/jenkins-build/build/workspace/ceph-perf-crimson/ceph-master/build/lib/libcls_journal.so (dlopen failed): /home/jenkins-build/build/workspace/ceph-perf-crimson/ceph-master/build/lib/libcls_journal.so: undefined symbol: _Z25cls_get_pool_stripe_widthPv
```
Explanation:
1) Some application use dlopen to load library libcls_journal.so
2) cls_get_pool_stripe_width is declared in objclass/objclass.h
3) cls_get_pool_stripe_width is defined in osd/objclass.cc
4) osd/objclass.cc isn't built into libcls_journal.so

**Pull request closed without merging**